qiskit_nature/results/utils/protein_shape_decoder.py

```diff
@@ -0,0 +1,129 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""An auxiliary class that gets the turns in the main and side chain of a molecule
+ in :class:`ProteinFoldingResult`."""
+from typing import List, Tuple, Optional
+
+
+class ProteinShapeDecoder:
+
+    """This class handles the decoding of the compact solution in :class:`ProteinFoldingProblem`
+    and returns the information encoded in the result about the turns
+    associated to the main and side chains.
+    """
```

Comment:
Normally the intent is to try an have a brief sentence overview and then more description. Bear in mind the first sentence too is what is displayed in the index tables in the documentation so it should not be too much. Maybe this class is not part of the public API we have in the documentation. Its worth bearing in mind though.